sdk/Lusid.Sdk/Utilities/LusidExceptionHandler.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using Lusid.Sdk.Client;
+
+namespace Lusid.Sdk.Utilities
+{
+    /// <summary>
+    /// Handles the generation of LUSID exceptions after receiving the ApiResponse
+    /// </summary>
+    internal static class LusidExceptionHandler
+    {
+        /// <summary>
+        /// Generate exceptions from the ApiResponse when ResponseStatus is an Error,
+        /// and StatusCode has no content or is less than 400
+        /// </summary>
+        /// <param name="methodName">The name of the method that was being called</param>
+        /// <param name="response">The ApiResponse</param>
+        public static Exception CustomExceptionFactory(string methodName, IApiResponse response)
+        {
+            // Use default exception handler first (only use subsequent checks if this returns null)
+            Exception defaultException = Configuration.DefaultExceptionFactory.Invoke(methodName, response);
+            if (defaultException != null)
+            {
+                return defaultException;
+            }
+
+            // Throw if ErrorText has been populated:
+            //  - Internal SDK deserialization errors will result in ErrorText to be not null.
+            //  - Network-level errors will also result in ErrorText being populated.
+            if (response.ErrorText != null)
+            {
+                return new ApiException(
+                    (int) response.StatusCode,
+                    $"Internal SDK error occurred when calling {methodName}: {response.ErrorText}",
+                    response.ErrorText,
+                    response.Headers
+                );
+            }
+
+            return null;
```

Comment:
Note, I haven't ported across the behaviour here:
https://github.com/finbourne/lusid-sdk-csharp-preview/pull/230/files#diff-d62b6c2b0eb55f9d07ccad692c877fd73ff302448dfab7697134a801f5cfe685R41

...which is the handling for null response body values (due to deserialisation issues). 

Are there any calls which legitimately do not return a response body (only an http status code) - e.g. PUTs / POSTs?

Reply:
It is an observed behaviour where the JsonConvert doesn't actually fail when we pass an empty string or whitespace here
https://github.com/finbourne/lusid-sdk-csharp-preview/blob/master/sdk/Lusid.Sdk/Client/ApiClient.cs#L146

It in turn returns the same empty string or whitespace. 

Considering this is rather strange, and should maybe be investigated deeper, I am happy for this to be left out for now. 
The comment for awareness on the `ErrorText` check should suffice

Reply:
That's because by default JsonConvert doesn't have `PropertyRequired` set by default

https://www.newtonsoft.com/json/help/html/JsonPropertyRequired.htm
https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_Required.htm